src/products/utils/validation.ts

```diff
@@ -1,5 +1,44 @@
+import {
+  ProductErrorCode,
+  ProductErrorWithAttributesFragment,
+} from "@saleor/graphql";
+
+import { ProductCreateData } from "../components/ProductCreatePage";
+
 export const validatePrice = (price: string) =>
   price === "" || parseInt(price, 10) < 0;
 
 export const validateCostPrice = (price: string) =>
   price !== "" && parseInt(price, 10) < 0;
+
+export const validateProductCreateData = (data: ProductCreateData) => {
+  let errors: ProductErrorWithAttributesFragment[] = [];
+
+  if (!data.productType) {
+    errors = [
+      ...errors,
+      {
+        __typename: "ProductError",
+        code: ProductErrorCode.REQUIRED,
+        field: "productType",
+        message: null,
+        attributes: [],
+      },
+    ];
+  }
+
+  if (!data.name) {
+    errors = [
+      ...errors,
+      {
+        __typename: "ProductError",
+        code: ProductErrorCode.REQUIRED,
+        field: "name",
+        message: null,
+        attributes: [],
+      },
+    ];
+  }
+
+  return errors;
+};
```

Comment:
Isn't the same?

```suggestion
const createEmptyRequiredError = (field: string) => ({
    __typename: "ProductError",
     code: ProductErrorCode.REQUIRED,
     field,
     message: null,
     attributes: [],
})

export const validateProductCreateData = (data: ProductCreateData) => {
  if (!data.productType) {
   return [createEmptyRequiredError('productType')]
  }

  if (!data.name) {
      return [createEmptyRequiredError('name')]
  }

  return [];
};
```

Why do you create an empty array (errors) and then merge everything else to it? Do you actually need that?

Reply:
Yes, it is the same, Ill change it